sklearn/utils/tests/test_optimize.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+
+from sklearn.utils.optimize import newton_cg
+from scipy.optimize import fmin_ncg
+
+from sklearn.utils.testing import assert_array_almost_equal
+
+
+def test_newton_cg():
+    # Test that newton_cg gives same result as scipy's fmin_ncg
+    rng = np.random.RandomState(0)
+    A = rng.normal(size=(10, 10))
+    x0 = np.ones(10)
+
+    def func(x):
+        Ax = A.dot(x)
+        return .5 * (Ax).dot(Ax)
+
+    def grad(x):
+        return A.T.dot(A.dot(x))
+
+    def hess(x, p):
+        return p.dot(A.T.dot(A.dot(x.all())))
+
+    def func_grad_hess(x):
+        return func(x), grad(x), lambda x: A.T.dot(A.dot(x))
```

Comment:
why is this different from ``hess``? It seems you use two different interfaces. if you use this here, maybe use fhess in ``fmin_ncg`` instead ``fhess_p``?